index.js

```diff
@@ -177,6 +180,11 @@ function compression (options) {
       var negotiator = new Negotiator(req)
       var method = negotiator.encoding(['gzip', 'deflate', 'identity'], ['gzip'])
 
+      // if no method is found, use the default encoding
+      if (encodingSupported.indexOf(enforceEncoding) !== -1 && !req.headers['accept-encoding']) {
+        method = enforceEncoding === '*' ? 'gzip' : enforceEncoding
```

Comment:
I dont see `*` referenced anywhere but here. Can you describe why this is necessary when you could drop this check and just have folks pass `gzip` directly for this?

Reply:
If * is not checked and changed, the deflate encoding would be applied as it’s the last one in the check (https://github.com/expressjs/compression/blob/66c0cb759d5ea36e6253df5f2fa88e4f6b6ef8c3/index.js#L196-L198), and deflate is not the encoding we want to be applied, it should be gzip when it’s *.